JPS_VIRTUALSENSOR/ShipInputAgent/src/main/java/com/cmclinnovations/ship/DataUploader.java

```diff
@@ -148,18 +202,22 @@ static List<Ship> uploadDataFromFile(QueryClient queryClient) throws IOException
                 LOGGER.error(e.getMessage());
             }
         }
-
-        uploadShips(ships, queryClient);
-
         return ships;
     }
 
-    static void updateFile(File file, String fileContent) {
-        try (FileOutputStream outputStream = new FileOutputStream(file)) {
-            outputStream.write(fileContent.getBytes());
-        } catch (Exception e) {
-            LOGGER.error(e.getMessage());
-        }
+    public static int loadDataFromRDB(QueryClient queryClient) throws IOException {
+        String sqlQuery = "SELECT DISTINCT points.\"MMSI\", points.\"VesselType\" AS \"SHIPTYPE\", 0 AS \"SPEED\", " + //
+                " 0.0 AS \"COURSE\", 0.0 AS \"LAT\", 0.0 AS \"LON\", '2024-01-01T12:00:00' AS \"TIMESTAMP\" " + //
+                "FROM (SELECT \"MMSI\", \"VesselType\", \"geom\" FROM \"ship\") AS points, " + //
+                "(SELECT \"geom\" FROM \"scopes\") AS polygon WHERE " + //
+                "ST_Contains(polygon.geom, points.geom)";
```

Comment:
looks like this relies on the scopes table being initialised which I'm not a big fan of. I personally think it's fine to upload everything unless there's performance issue?

Reply:
This is now explicitly explained in the README.